docs/ingestion/schema-design.md

```diff
@@ -241,12 +240,45 @@ You should query for the number of ingested rows with:
+- [String-based schema discovery](#string-based-schema-discovery) where all the discovered columns are typed as either native string or multi-value string columns.
+
+#### Type-aware schema discovery
+
+> Note that using type-aware schema discovery can impact downstream BI tools depending on how they handle ARRAY typed columns..
+
+You can have Druid infer the schema and types for your data partially or fully by setting `dimensionsSpec.useSchemaDiscovery` to `true` and defining some or no dimensions in the dimensions list. 
+
+When performing type-aware schema discovery, Druid can discover all of the columns of your input data (that aren't in the exclusion list), but the behavior depends on the source data:
+
+- For primitive types, Druid ingests them as the most appropriate native Druid type, either string, double or long. 
+- For input formats with native boolean types will be ingested as strings or longs depending on the value of `druid.expressions.useStrictBooleans`.
+- For arrays, the data gets ingested as an array typed column: `ARRAY<STRING>`, `ARRAY<DOUBLE>`, or `ARRAY<LONG>`. You can then interact with these columns using ARRAY functions as well as UNNEST.
+- For mixed types, complex nested structures, and arrays of complex nested structures, Druid ingests the column as the least restrictive data type.
+
+If you're already using string-based schema discovery and want to migrate, see [Migrating to type-aware schema discovery](#migrating-to-type-aware-schema-discovery).
+
+#### String-based schema discovery
+
+If you do not set `dimensionsSpec.useSchemaDiscovery` to `true`, Druid can still use the string-based schema discovery for ingestion if any of the following conditions are met: 
+
+- The dimension list is empty 
+- You set `includeAllDimensions` to `true` 
+
+Druid coerces primitives and arrays of primitive types into the native Druid string type. Nested data structures and arrays of nested data structures are ignored and not ingested.
+
+#### Migrating to type-aware schema discovery
 
-If the `dimensions` field is left empty in your ingestion spec, Druid will treat every column that is not the timestamp column,
-a dimension that has been excluded, or a metric column as a dimension.
+If you previously used string-based schema discovery and want to migrate to type-aware schema discovery, you'll need to do the following:
 
-Note that when using schema-less ingestion, all dimensions will be ingested as String-typed dimensions.
+- Update any queries that use multi-value dimensions (MVDs) to use UNNEST in conjunction with other functions so that no  MVD behavior is being relied upon. Type-aware schema discovery generates ARRAY typed columns instead of MVDs, so queries that use any MVD features will fail.
```

Comment:
```suggestion
- Update any queries that use multi-value dimensions (MVDs) to use UNNEST in conjunction with other functions so that no MVD behavior is being relied upon. Type-aware schema discovery generates ARRAY typed columns instead of MVDs, so queries that use any MVD features will fail.
```